src/mantine-core/src/components/Select/Select.tsx

```diff
@@ -214,6 +214,10 @@ export const Select = forwardRef<HTMLInputElement, SelectProps>(
     const isDeselectable = allowDeselect === undefined ? clearable : allowDeselect;
 
     const setDropdownOpened = (opened: boolean) => {
+      if (dropdownOpened === opened) {
```

Comment:
Please change code a little to make it consistent with the rest of codebase: do not use early returns, replace it with if condition istead:

```ts
if (dropdownOpened !== opened) {
      _setDropdownOpened(opened);
      const handler = opened ? onDropdownOpen : onDropdownClose;
      typeof handler === 'function' && handler();
}
```